packages/api/portfolios/environments/createEnvironment.ts

```diff
@@ -0,0 +1,109 @@
+import "reflect-metadata";
+import { createConnection } from "../../utils/database";
+import { Portfolio } from "../../../orm/entity/Portfolio";
+import { Application } from "../../../orm/entity/Application";
+import { Environment, IEnvironment, IEnvironmentCreate } from "../../../orm/entity/Environment";
+import { EnvironmentRepository } from "../../repository/EnvironmentRepository";
+import { APIGatewayProxyResult, Context } from "aws-lambda";
+import { ApiSuccessResponse, SuccessStatusCode } from "../../utils/response";
+import internalSchema = require("../../models/internalSchema.json");
+import middy from "@middy/core";
+import xssSanitizer from "../../portfolioDrafts/xssSanitizer";
+import jsonBodyParser from "@middy/http-json-body-parser";
+import validator from "@middy/validator";
+import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
+import cors from "@middy/http-cors";
+import { ApiGatewayEventParsed } from "../../utils/eventHandlingTool";
+import { CORS_CONFIGURATION } from "../../utils/corsConfig";
+import { wrapSchema } from "../../utils/schemaWrapper";
+import { errorHandlingMiddleware } from "../../utils/errorHandlingMiddleware";
+import createError from "http-errors";
+import { IpCheckerMiddleware } from "../../utils/ipLogging";
+import { validateRequestShape } from "../../utils/shapeValidator";
+
+/**
+ * Submits the environment of an application
+ *
+ * @param event - The POST request from API Gateway
+ */
+export async function baseHandler(
+  event: ApiGatewayEventParsed<IEnvironment>,
+  context?: Context
+): Promise<APIGatewayProxyResult> {
+  const setupResult = validateRequestShape<IEnvironment>(event);
+  const { portfolioId, applicationId } = setupResult.path;
+  const environmentBody = setupResult.bodyObject;
+
+  // set up database connection
+  const connection = await createConnection();
+  const insertedEnvironments: IEnvironmentCreate[] = [];
+
+  try {
+    // ensures both the portfolio and application exists
+    await connection.getRepository(Portfolio).findOneOrFail({
+      id: portfolioId,
+    });
+    const application = await connection.getRepository(Application).findOneOrFail({
+      id: applicationId,
+    });
+
+    // get all environment names for application
+    const environments = await connection
+      .getRepository(Environment)
+      .createQueryBuilder("environment")
+      .select(["environment.name"])
+      .where("environment.applicationId = :applicationId", { applicationId: application.id })
+      .getMany();
+
+    // ensure the environment name is unique for the application
+    // according to business rule 3.3
+    for (const environment of environments) {
+      if (environment.name === environmentBody.name) {
+        throw createError(400, "Duplicate Environment name in application", {
+          errorName: "DuplicateEnvironmentName",
+          environmentName: environmentBody.name,
+        });
+      }
+    }
```

Comment:
We should pull this into Environment repository, we need this for the UPDATE and we need it for Application operations.

Reply:
Added a method to get all the env names in the Environment repository to assist with that. You can maybe do an .include() on names to see if the name is taken to simplify the code.